src/test/ui/issues/issue-21974.stderr

```diff
@@ -2,7 +2,7 @@ error[E0283]: type annotations needed
   --> $DIR/issue-21974.rs:11:19
    |
 LL | trait Foo {
-   | --------- required by `Foo`
+   | --------- required by this bound in `Foo`
 ...
 LL |     where &'a T : Foo,
    |                   ^^^ cannot infer type for reference `&'a T`
```

Comment:
There are a few cases like this one where the inference error itself is already incorrect and the wording change makes them slightly worse in my eyes.

They seem rare enough that we don't need to deal with them immediately, but we should probably account for them.